frontend/src/routes/project/[project_code]/+page.ts

```diff
@@ -1,147 +1,192 @@
-
-import type { PageLoadEvent } from "./$types";
-import { getClient } from "$lib/graphQLClient";
-import { graphql } from "$lib/gql";
-import { invalidate } from "$app/navigation";
+import type {
+  AddProjectMemberInput,
+  ChangeProjectDescriptionInput,
+  ChangeProjectMemberRoleInput,
+  ChangeProjectNameInput,
+  ProjectPageQuery,
+} from '$lib/gql/graphql';
 
+import { invalidate } from '$app/navigation';
+import { graphql } from '$lib/gql';
+import { getClient } from '$lib/graphQLClient';
+import type { PageLoadEvent } from './$types';
 import logsample from './logsample.json';
 
-export type ProjectUser = ProjectPageQuery["projects"][0]["ProjectUsers"][number];
+export type ProjectUser = ProjectPageQuery['projects'][0]['ProjectUsers'][number];
 
 export async function load(event: PageLoadEvent) {
-    const client = getClient(event);
-    const projectCode = event.params.project_code;
-    const result = await client.query(graphql(`
-        query projectPage($projectCode: String!) {
-            projects(where: {code: {_eq: $projectCode}}) {
-                id
-                name
-                code
-                description
-                type
-                lastCommit
-                createdDate
-                retentionPolicy
-                ProjectUsers {
-                    id
-                    role
-                    User {
-                        id
-                        name
-                    }
-                }
-                changesets {
-                    node
-                    parents
-                    date
-                    user
-                    desc
-                }
-            }
-        }
-`), { projectCode }).toPromise();
-    if (result.error) throw new Error(result.error.message);
-    event.depends(`project:${result.data?.projects[0]?.id}`);
-    return {
-        project: result.data?.projects[0],
-        log: logsample,
-        code: projectCode
-    };
+	const client = getClient(event);
+	const projectCode = event.params.project_code;
+	const result = await client
+		.query(
+			graphql(`
+				query projectPage($projectCode: String!) {
+					projects(where: { code: { _eq: $projectCode } }) {
+						id
+						name
+						code
+						description
+						type
+						lastCommit
+						createdDate
+						retentionPolicy
+						ProjectUsers {
+							id
+							role
+							User {
+								id
+								name
+							}
+						}
+						changesets {
+							node
+							parents
+							date
+							user
+							desc
+						}
+					}
+				}
+			`),
+			{ projectCode },
+		)
+		.toPromise();
+	if (result.error) throw new Error(result.error.message);
+	event.depends(`project:${result.data?.projects[0]?.id}`);
+	return {
+		project: result.data?.projects[0],
+		log: logsample,
+		code: projectCode,
+	};
+}
+
+export async function _addProjectMember(input: AddProjectMemberInput) {
```

Comment:
I'm still not crazy about exporting stuff from here that not part of the convention for this file type...

Reply:
👍 we'll discuss this [soon](https://github.com/sillsdev/languageforge-lexbox/pull/65#discussion_r1206322550)